dll/cpl/sysdm/environment.c

```diff
@@ -345,43 +1099,48 @@ OnNewVariable(HWND hwndDlg,
               INT iDlgItem)
 {
     HWND hwndListView;
-    PVARIABLE_DATA VarData;
+    PDIALOG_DATA DlgData;
     LV_ITEM lvi;
     INT iItem;
 
+    DlgData = GlobalAlloc(GPTR, sizeof(DIALOG_DATA));
+
     hwndListView = GetDlgItem(hwndDlg, iDlgItem);
 
-    VarData = GlobalAlloc(GPTR, sizeof(VARIABLE_DATA));
+    DlgData->bIsFriendlyUI = FALSE;
+    DlgData->dwSelectedValueIndex = -1;
+
+    DlgData->VarData = GlobalAlloc(GPTR, sizeof(VARIABLE_DATA));
 
     if (DialogBoxParam(hApplet,
                        MAKEINTRESOURCE(IDD_EDIT_VARIABLE),
                        hwndDlg,
                        EditVariableDlgProc,
-                       (LPARAM)VarData) <= 0)
+                       (LPARAM)DlgData) <= 0)
     {
-        if (VarData->lpName != NULL)
-            GlobalFree(VarData->lpName);
+        if (DlgData->VarData->lpName != NULL)
+            GlobalFree(DlgData->VarData->lpName);
 
-        if (VarData->lpRawValue != NULL)
-            GlobalFree(VarData->lpRawValue);
+        if (DlgData->VarData->lpRawValue != NULL)
+            GlobalFree(DlgData->VarData->lpRawValue);
 
-        if (VarData->lpCookedValue != NULL)
-            GlobalFree(VarData->lpCookedValue);
+        if (DlgData->VarData->lpCookedValue != NULL)
+            GlobalFree(DlgData->VarData->lpCookedValue);
 
-        GlobalFree(VarData);
+        GlobalFree(DlgData);
     }
     else
     {
-        if (VarData->lpName != NULL && (VarData->lpCookedValue || VarData->lpRawValue))
+        if (DlgData->VarData->lpName != NULL && (DlgData->VarData->lpCookedValue || DlgData->VarData->lpRawValue))
         {
             memset(&lvi, 0x00, sizeof(lvi));
```

Comment:
```suggestion
            ZeroMemory(&lvi, sizeof(lvi));
```
and same suggestion for the similar line in OnEditVariable().